trieste/objectives/multi_objectives.py

```diff
@@ -55,39 +55,49 @@ class MultiObjectiveTestProblem(ObjectiveTestProblem):
     random number seed."""
 
 
-def vlmop2(x: TensorType) -> TensorType:
+def vlmop2(x: TensorType, d: int) -> TensorType:
     """
     The VLMOP2 synthetic function.
 
-    :param x: The points at which to evaluate the function, with shape [..., 2].
+    :param x: The points at which to evaluate the function, with shape [..., d].
+    :param d: The dimensionality of the synthetic function.
     :return: The function values at ``x``, with shape [..., 2].
     :raise ValueError (or InvalidArgumentError): If ``x`` has an invalid shape.
     """
-    tf.debugging.assert_shapes([(x, (..., 2))], message="vlmop2 only allow 2d input")
-    transl = 1 / math.sqrt(2.0)
+    tf.debugging.assert_shapes(
+        [(x, (..., d))],
+        message=f"input x dim: {x.shape[-1]} is not align with pre-specified dim: {d}",
+    )
+    transl = 1 / tf.sqrt(tf.cast(d, x.dtype))
     y1 = 1 - tf.exp(-1 * tf.reduce_sum((x - transl) ** 2, axis=-1))
     y2 = 1 - tf.exp(-1 * tf.reduce_sum((x + transl) ** 2, axis=-1))
     return tf.stack([y1, y2], axis=-1)
 
 
-def vlmop2_gen_pareto_optimal_points(n: int, seed: int | None = None) -> TensorType:
-    tf.debugging.assert_greater(n, 0)
-    _x = tf.linspace([-1 / tf.sqrt(2.0)], [1 / tf.sqrt(2.0)], n)
-    return vlmop2(tf.concat([_x, _x], axis=1))
+def VLMOP2(input_dim: int) -> MultiObjectiveTestProblem:
```

Comment:
I assume the 2 in VLMOP2 doesn't refer to the dimensions then?

Reply:
it does, dimensionality of the search space... why do you say that?